servers/republik/modules/crowdfundings/graphql/resolvers/_mutations/resetMembership.js

```diff
@@ -0,0 +1,73 @@
+const { Roles } = require('@orbiting/backend-modules-auth')
+const { publishMonitor } = require('../../../../../lib/slack')
+const { canReset } = require('../Membership')
+
+module.exports = async (_, args, context) => {
+  const { pgdb, req, t, user: me, mail: { enforceSubscriptions } } = context
+  Roles.ensureUserHasRole(me, 'supporter')
+
+  const { id: membershipId } = args
+
+  const membership = await pgdb.public.memberships.findOne({ id: membershipId })
+  if (!membership) {
+    throw new Error(t('api/membership/404'))
+  }
+
+  const transaction = await pgdb.transactionBegin()
+
+  try {
+    await canReset(
+      membership,
+      { throwError: true },
+      { ...context, pgdb: transaction }
+    )
+
+    await transaction.public.membershipPeriods.delete({ membershipId })
+
+    const pledge = await transaction.public.pledges.findOne({ id: membership.pledgeId })
+
+    const voucherCode = await transaction.queryOneField(
+      'SELECT make_hrid(\'"memberships"\'::regclass, \'voucherCode\'::text, 6::bigint)'
+    )
+
+    const updatedMembership = await transaction.public.memberships.updateAndGetOne(
+      { id: membershipId },
+      {
+        active: false, // disable membership
+        userId: pledge.userId, // return membership to pledger
+        voucherCode, // generate new voucher code
+        voucherable: true,
+        updatedAt: new Date()
+      }
+    )
+
+    await transaction.transactionCommit()
+
+    const { active, id, sequenceNumber, userId } = updatedMembership
+
+    try {
+      if (active) {
```

Comment:
why would active ever be true? The `updateAndGetOne` explicitly sets active to `false`. Should the `enforceSubscriptions` be on the original owner (`membership.userId`)? (in cases that one no longer has an active membership and needs updated customer record on MC)

Reply:
This can't ever be true, this is a horribly piece of code. Changing.